api/app/organisation/resolver.py

```diff
@@ -0,0 +1,23 @@
+from app.organisation.repository import OrganisationRepository
+from app import LOGGER
+
+class OrganisationResolver():
+    _cache = None
+    _default = None
+
+    @classmethod
+    def resolve_from_domain(cls, domain):
+        if not cls._cache:
+            LOGGER.info('Populating Organisation Cache')
+            organisations = OrganisationRepository.get_all()
+            cls._cache = {}
+            for org in organisations:
+                cls._cache[org.domain] = org
+                if org.is_default:
+                    cls._default = org
+        
+        if domain not in cls._cache:
+            LOGGER.warning('Could not find domain {} in cache, returning default organisation'.format(domain))
+            return cls._default
```

Comment:
Should this return the default org? If for some reason an organisation isn't found, my feeling is that the site should break or at the very least return some user friendly error page (and the log should be at an error level since obviously something has gone really wrong). With the way it is now, the Deep Learning Indaba site will display potentially causing a lot of confusion to any users that might be there expecting some other organisation.

Reply:
Yeah that's a good point! Have updated to remove the default option and make failure to resolve the organisation and error. Thanks!

Reply:
Ah damn, now without the default organisation, all the tests are broken because they don't set the origin header 😂